src/Shell/Task/ModelTask.php

```diff
@@ -344,25 +344,56 @@ public function findHasMany($model, array $associations)
                 }
             }
 
-            foreach ($otherSchema->columns() as $fieldName) {
-                $assoc = false;
-                if (!in_array($fieldName, $primaryKey) && $fieldName === $foreignKey) {
-                    $assoc = [
-                        'alias' => $otherModel->alias(),
-                        'foreignKey' => $fieldName
-                    ];
-                } elseif ($otherTableName === $tableName && $fieldName === 'parent_id') {
-                    $className = ($this->plugin) ? $this->plugin . '.' . $model->alias() : $model->alias();
-                    $assoc = [
-                        'alias' => 'Child' . $model->alias(),
-                        'className' => $className,
-                        'foreignKey' => $fieldName
-                    ];
+            $targetForeignKey = $primaryKey;
+            foreach ($targetForeignKey as $i => $fieldName) {
+                if ($fieldName === 'id') {
+                    $targetForeignKey[$i] = Inflector::singularize($model->table()) . '_id';
                 }
-                if ($assoc && $this->plugin && empty($assoc['className'])) {
-                    $assoc['className'] = $this->plugin . '.' . $assoc['alias'];
+            }
+
+            $possibleForeignKeys = $otherSchema->columns();
+            foreach ($possibleForeignKeys as $i => $fieldName) {
+                if ($fieldName === 'parent_id') {
+                    continue;
+                }
+                if (strlen($fieldName) <= 3 || !preg_match('/^.*_id$/', $fieldName)) {
+                    unset($possibleForeignKeys[$i]);
                 }
-                if ($assoc) {
+            }
+
+            $assoc = false;
+            if (Hash::diff(
+                array_values($targetForeignKey), array_values($possibleForeignKeys)) === []
```

Comment:
I did this to confirm your suspicion, right above L365:
```php
debug([
    '$targetForeignKey' => array_values($targetForeignKey),
    '$possibleForeignKeys' => array_values($possibleForeignKeys),
    'Hash::diff' => Hash::diff(
        array_values($targetForeignKey), array_values($possibleForeignKeys))
]);
```

Ran `$ bin/cake bake model tweets --force`

# Result
```bash
[01:05 PM]-[vagrant@project]-[/var/www/app]-[git master] 
$ bin/cake bake model tweets --force

Welcome to CakePHP v3.2.8 Console
---------------------------------------------------------------
App : src
Path: /var/www/app/src/
PHP : 7.0.5-3+donate.sury.org~trusty+1
---------------------------------------------------------------
One moment while associations are detected.
/vendor/cakephp/bake/src/Shell/Task/ModelTask.php (line 367)
########## DEBUG ##########
[
	'$targetForeignKey' => [
		(int) 0 => 'project_id',
		(int) 1 => 'tweet_id'
	],
	'$possibleForeignKeys' => [
		(int) 0 => 'project_id',
		(int) 1 => 'user_id',
		(int) 2 => 'tweet_id'
	],
	'Hash::diff' => [
		(int) 1 => 'tweet_id',
		(int) 2 => 'tweet_id'
	]
]
###########################
/vendor/cakephp/bake/src/Shell/Task/ModelTask.php (line 367)
########## DEBUG ##########
[
	'$targetForeignKey' => [
		(int) 0 => 'project_id',
		(int) 1 => 'tweet_id'
	],
	'$possibleForeignKeys' => [
		(int) 0 => 'project_id',
		(int) 1 => 'user_id',
		(int) 2 => 'tweet_id'
	],
	'Hash::diff' => [
		(int) 1 => 'tweet_id',
		(int) 2 => 'tweet_id'
	]
]
###########################
/vendor/cakephp/bake/src/Shell/Task/ModelTask.php (line 367)
########## DEBUG ##########
[
	'$targetForeignKey' => [
		(int) 0 => 'project_id',
		(int) 1 => 'tweet_id'
	],
	'$possibleForeignKeys' => [
		(int) 0 => 'project_id'
	],
	'Hash::diff' => [
		(int) 1 => 'tweet_id'
	]
]
###########################
/vendor/cakephp/bake/src/Shell/Task/ModelTask.php (line 367)
########## DEBUG ##########
[
	'$targetForeignKey' => [
		(int) 0 => 'project_id',
		(int) 1 => 'tweet_id'
	],
	'$possibleForeignKeys' => [
		(int) 0 => 'project_id',
		(int) 1 => 'group_id',
		(int) 2 => 'user_id'
	],
	'Hash::diff' => [
		(int) 1 => 'tweet_id',
		(int) 2 => 'user_id'
	]
]
###########################
/vendor/cakephp/bake/src/Shell/Task/ModelTask.php (line 367)
########## DEBUG ##########
[
	'$targetForeignKey' => [
		(int) 0 => 'project_id',
		(int) 1 => 'tweet_id'
	],
	'$possibleForeignKeys' => [
		(int) 0 => 'project_id',
		(int) 1 => 'user_id'
	],
	'Hash::diff' => [
		(int) 1 => 'tweet_id'
	]
]
###########################
/vendor/cakephp/bake/src/Shell/Task/ModelTask.php (line 367)
########## DEBUG ##########
[
	'$targetForeignKey' => [
		(int) 0 => 'project_id',
		(int) 1 => 'tweet_id'
	],
	'$possibleForeignKeys' => [],
	'Hash::diff' => [
		(int) 0 => 'project_id',
		(int) 1 => 'tweet_id'
	]
]
###########################
/vendor/cakephp/bake/src/Shell/Task/ModelTask.php (line 367)
########## DEBUG ##########
[
	'$targetForeignKey' => [
		(int) 0 => 'project_id',
		(int) 1 => 'tweet_id'
	],
	'$possibleForeignKeys' => [
		(int) 0 => 'project_id',
		(int) 1 => 'user_id'
	],
	'Hash::diff' => [
		(int) 1 => 'tweet_id'
	]
]
###########################
/vendor/cakephp/bake/src/Shell/Task/ModelTask.php (line 367)
########## DEBUG ##########
[
	'$targetForeignKey' => [
		(int) 0 => 'project_id',
		(int) 1 => 'tweet_id'
	],
	'$possibleForeignKeys' => [
		(int) 0 => 'project_id'
	],
	'Hash::diff' => [
		(int) 1 => 'tweet_id'
	]
]
###########################

Baking table class for Tweets...

Creating file /var/www/app/src/Model/Table/TweetsTable.php
Wrote `/var/www/app/src/Model/Table/TweetsTable.php`

Baking entity class for Tweet...

Creating file /var/www/app/src/Model/Entity/Tweet.php
Wrote `/var/www/app/src/Model/Entity/Tweet.php`

Baking test fixture for Tweets...

Creating file /var/www/app/tests/Fixture/TweetsFixture.php
Wrote `/var/www/app/tests/Fixture/TweetsFixture.php`
Bake is detecting possible fixtures...

Baking test case for App\Model\Table\TweetsTable ...

Creating file /var/www/app/tests/TestCase/Model/Table/TweetsTableTest.php
Wrote `/var/www/app/tests/TestCase/Model/Table/TweetsTableTest.php`

[01:05 PM]-[vagrant@project]-[/var/www/app]-[git master] 
$ 
```